### PR TITLE
Work2

### PR DESCRIPTION
**Pull request by DhruvWalia03**
**Fixes** #23 

[//]:  #23 
[//]: # changed the club fragment and detail sheet

[//]: # (**ScreenShots**)

### CODE REVIEW

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import kotlinx.android.synthetic.main.fragment_club_detail.view.club_name
+import kotlinx.android.synthetic.main.fragment_club_detail.view.club_web
+import kotlinx.android.synthetic.main.fragment_club_detail.view.club_youtube
+import kotlinx.android.synthetic.main.fragment_club_detail.view.web
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.IOException
+import java.nio.charset.StandardCharsets
+
+@Suppress("DEPRECATION")
```
Comment (yash1ts) [inline app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:30]: What is deprecated? Can it be replaced with the newer method?

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+class ClubDetailBottomSheet : BottomSheetDialogFragment() {
+
+    private var i = 0
+    private var it = 0
```
Comment (yash1ts) [inline app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:34]: give better names

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    private var chromeTab: ChromeTab? = null
```
Comment (yash1ts) [inline app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:35]: need not to be null

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+    @SuppressLint("ClickableViewAccessibility")
```
Comment (yash1ts) [inline app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:37]: Check if we can clear this warning

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.fragment_club_detail, container, false)
+        chromeTab = ChromeTab(context)
+        view.web.isHorizontalScrollBarEnabled = true
+        view.club_icon.hierarchy.setProgressBarImage(CircularProgressDrawable((context)!!))
+        if (BuildConfig.DEBUG && arguments == null) {
+            error("Assertion failed")
+        }
+        i = requireArguments().getInt("club_id")
```
Comment (yash1ts) [inline app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:46]: make constant for club_id

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+        getJson(view)
+        return view
+    }
+
+    private fun getJson(view: View) {
```
Comment (yash1ts) [inline app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:51]: getJson() should just get json. Do other things in a separate function.

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+        val json: String
+        try {
+            val inputStream = (context)!!.assets.open("clubs.json")
+            val size = inputStream.available()
+            val buffer = ByteArray(size)
+            inputStream.read(buffer)
+            inputStream.close()
+            json = String(buffer, StandardCharsets.UTF_8)
+            val jsonObject = JSONObject(json)
```
Comment (yash1ts) [inline app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:60]: All the above things can be done in a better way please see it's already used in contact activity

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+            val `object` = jsonObject.getJSONArray("modelList").getJSONObject(i - 1)
```
Comment (yash1ts) [inline app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:61]: object is a reserved word just use some other name instead.

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+                13 -> it = R.drawable.kartavya_logo
+                14 -> it = R.drawable.rythm_logo
+                15 -> it = R.drawable.litm_logo
+                16 -> it = R.drawable.lci_logo
+                17 -> it = R.drawable.ff_logo
+                18 -> it = R.drawable.roboism_logo
+                19 -> it = R.drawable.manthan_logo
+                20 -> it = R.drawable.adc_logo
+                21 -> it = R.drawable.art_logo
+            }
```
Comment (yash1ts) [inline app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:84]: just get them in an array then maybe you can use a for loop to assign them.

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+//                  "\t<div id='yeep' style='width: 100%;height:100%;text-align:center' >\n" +
+//                  "\t</div>\n" +
+//                  "</body>\n" +
+//                  "\n" +
+//                  "<script type=\"text/javascript\">\n" +
+//                  "\tvar s = document.getElementById(\"yeep\");\n" +
+//                  "\ts.innerHTML+=\"<iframe src='https://www.facebook.com/plugins/page.php?href=https%3A%2F%2Fwww.facebook.com%2F"+site+"&tabs=timeline&width=\"+screen.width+\"&height=\"+screen.height+\"&small_header=true&adapt_container_width=true&hide_cover=true&show_facepile=false&appId=384900825472866'  style='border:none;height:100%;width: 100%; ' scrolling='yes' frameborder='0' allowTransparency='true' allow='encrypted-media'></iframe>\";\n" +
+//                  "\t//document.write(s.innerHTML);\n" +
+//                  "</script>\n" +
+//                  "</html>\n";
```
Comment (yash1ts) [inline app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:137]: remove all these commented lines.

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+            "$site&tabs=timeline&width=350&height=700&small_header=true&adapt_container_width=true&hide_cover=true&" +
+            "show_facepile=false&appId=384900825472866\" width=\"100%\" height=\"100%\" style=\"border:none;" +
+            "overflow:hidden\" scrolling=\"no\" frameborder=\"0\" allowTransparency=\"true\"" +
+            " allow=\"encrypted-media\"></iframe>"
+        view.web.loadHtml(s)
+        // }
+        // });
+        view.web.setOnTouchListener { _: View?, event: MotionEvent ->
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) event.action == ACTION_BUTTON_PRESS else {
+            TODO("VERSION.SDK_INT < M")
```
Comment (yash1ts) [inline app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:148]: This piece of code was to restrict the user from clicking the webview. See if it works properly after removing the if condition.

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt (hunk 1 above)
Comment (DhruvWalia03) [reply app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:30]: setImageResource is deprecated. cant be replace even if I replace it shows the new one to be deprecated too

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt (hunk 7 above)
Comment (DhruvWalia03) [reply app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:60]: I was earlier trying the same thing as in club activity sir but it was giving errors when converted to kotlin
so I found this easy

File: app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt (hunk 11 above)
Comment (DhruvWalia03) [reply app/src/main/java/com/mailerdaemon/app/clubs/ClubDetailBottomSheet.kt:148]: warning show kr rha hai sir bss

**Comment by DhruvWalia03** [comment]
Sir i guess INTRO Activity me kuch dikaat hai .. woh sharedprefrence ki value ko false ni kr rha skip ya finish click krne pe

**Comment by yash1ts** [comment]
Yes, I saw that, it's fixed in the new branch new_update. Pull update from there and check that out.

**Pull request closed without merging**